trestle/tasks/tanium_to_oscal.py

```diff
@@ -0,0 +1,152 @@
+        
+    def simulate(self) -> TaskOutcome:
+        """Provide a simulated outcome."""
+        return self._transform(True)
+        
+    def execute(self) -> TaskOutcome:
+        """Provide an actual outcome."""
+        return self._transform(False)
+    
+    def _transform(self, simulate:bool=False):
+        mode = ''
+        if simulate:
+            mode = 'simulated-'
+        if not self._config:
+            logger.error(f'config missing')
+            return TaskOutcome(mode+'failure') 
+        # process config
+        idir = self._config.get('input-dir')
+        if idir is None:
+            logger.error(f'config missing "input-dir"')
+            return TaskOutcome(mode+'failure')
+        ipth = pathlib.Path(idir)
+        odir = self._config.get('output-dir')
+        if odir is None:
+            logger.error(f'config missing "output-dir"')
+            return TaskOutcome(mode+'failure')
+        opth = pathlib.Path(odir)
+        overwrite = self._config.getboolean('output-overwrite', True)
+        quiet = self._config.getboolean('quiet', False)
+        # timestamp
+        timestamp = self._config.get('timestamp')
+        if timestamp is not None:
+            try:
+                tanium.Rule.set_default_datetime(timestamp)
+            except Exception as e:
+                logger.error(f'config invalid "timestamp"')
+                return TaskOutcome(mode+'failure')  
+        # insure output folder exists
+        opth.mkdir(exist_ok=True, parents=True)
+        # examine each file in the input folder
+        for ifile in sorted(ipth.iterdir()):
+            # assemble collection comprising output file name to unprocessed content
+            collection = self._assemble(ifile)
+            observations, analysis = tanium.get_observations(collection, 'json')
+            oname = ifile.with_suffix('').name+'.oscal'+'.json'
+            ofile = opth / pathlib.Path(oname)
+            # only allow writing output file if either:
+            # a) it does not already exist, or
+            # b) output-overwrite flag is True
+            if not overwrite:
+                if ofile.exists():
+                    logger.error(f'file exists: {ofile}')
+                    return TaskOutcome(mode+'failure')
+            if not simulate:
+                if not quiet:
+                    logger.info(f'create: {ofile}')
+                # write the OSCAL to the output file
+                write_file = pathlib.Path(ofile).open('w', encoding=const.FILE_ENCODING)
+                write_file.write(observations)
+                # display analysis
+                if not quiet:
+                    logger.info(f'Rules Analysis:')
+                    logger.info(f'rules [dispatched]: {analysis["dispatched_rules"]}')
+                    logger.info(f'rules [unique]: {analysis["unique_rules"]}')
+                    logger.info(f'results: {analysis["results"]}')
+        return TaskOutcome(mode+'success')
```

Comment:
This is super minor, however, some of the operations are missing the standard spaces from a python perspective (like line 134).

Reply:
Fixed. Not sure why make code-format, code-lint don't fix/complain in the trestle project. I copied module to arboretum project and this problem was detected and fixed there.